api/src/modules/proxy/proxy.controller.ts

```diff
@@ -67,13 +66,98 @@ export class ProxyController {
     required: false,
     example: 'BRA.1',
   })
+  @Get('/administrative-areas/:level/preview/tiles/:z/:x/:y.mvt')
+  async proxyAdminAreaTile(@Req() request: Request, @Res() response: Response) {
+    return this.proxyService.proxyTileRequest(request, response);
+  }
+
+  @ApiOperation({
+    description: 'Get tile for protected areas.',
+  })
   /**
-   *
-   *@todo parse level from admin areas entity
-   *@todo add level validation here in proxy service. Duplicated on the geoprocessing api.
+   *@todo Change ApiOkResponse mvt type
    */
-  @Get('/administrative-areas/:level/preview/tiles/:z/:x/:y.mvt')
-  async proxyTile(@Req() request: Request, @Res() response: Response) {
+  @ApiOkResponse({
+    type: 'mvt',
+  })
+  @ApiUnauthorizedResponse()
+  @ApiForbiddenResponse()
+  @ApiParam({
+    name: 'z',
+    description: 'The zoom level ranging from 0 - 20',
+    type: Number,
+    required: true,
+  })
+  @ApiParam({
+    name: 'x',
+    description: 'The tile x offset on Mercator Projection',
+    type: Number,
+    required: true,
+  })
+  @ApiParam({
+    name: 'y',
+    description: 'The tile y offset on Mercator Projection',
+    type: Number,
+    required: true,
+  })
+  @ApiQuery({
+    name: 'wdpaid',
+    description: 'Id of WDPA area',
+    type: Number,
+    required: false,
+    example: '5555697751',
+  })
```

Comment:
which id are we after here? My understanding is that we may not have data in the `wdpa.wdpaid` column in the geodb for protected areas that are uploaded by users - should we use the `wdpa.id` UUID column instead? (moreover, right now `wdpaid` is nullable, whereas `id` is a PK)

Reply:
that was @aagm suggestion and based on what we have in our endpoint definition doc. I guess if we are going to have the case of `wdpa.wdpaid` with no data for the areas uploaded by the user, I'm happy to change this param to `wdpa.id`.

@aagm what do you think?

Reply:
yes i was willing to use based on wdpaid, but i didn't though on user uploaded data, so changing it based on the db based id wdpa.id is ok from my point of view!